sec1_intro_and_setup/test_app/config/initializers/application_controller_renderer.rb

```diff
@@ -0,0 +1,8 @@
+# Be sure to restart your server when you modify this file.
```

Comment:
Whats this file for?

Reply:
This file allows controllers to render outside their scope.

Reply:
What do you mean allows controller to render outside their scope?

Reply:
Render an arbitrary template (`example.org` in this file) without being in the controller actions.

Reply:
can you elaborate?

Reply:
For example, if your view template needs to create an URL, then it will refer to this file to know what hostname to use and whether to use SSL, by default, the HTTP is `example.org` and SSL is not use (`https: false`), so your view template, let say call `test` will have the URL: `http://example.org/test`.

Note: Secure Sockets Layer (SSL), a protocol that enables secure communications online by encrypting the link between server and client.

Reply:
> if your view template needs to create an URL, then it will refer to this file to know what hostname to use and whether to use SSL

How does this happen?

Reply:
The `ActionController::Renderer` will render the view template with the absence of controller actions. It allows you to call a method called `render` where you specify which template to render.